framework_tvb/tvb/interfaces/web/controllers/flow_controller.py

```diff
@@ -377,64 +383,40 @@ def accepted__aggregation_functions():
-        if constants.ELEM_CONDITIONS in current_node:
-            new_filter = copy.deepcopy(current_node[constants.ELEM_CONDITIONS])
-        else:
-            new_filter = FilterChain()
-        new_filter.fields.extend(filters[FILTER_FIELDS])
-        new_filter.operations.extend(filters[FILTER_OPERATIONS])
-        new_filter.values.extend(filters[FILTER_VALUES])
-        # Get dataTypes that match the filters from DB then populate with values
-        values, total_count = [], 0
-        # Create a dictionary that matches what the template expects
-        parameters = {ABCAdapter.KEY_NAME: name,
-                      ABCAdapter.KEY_FILTERABLE: availablefilter,
-                      ABCAdapter.KEY_TYPE: constants.TYPE_SELECT,
-                      ABCAdapter.KEY_OPTIONS: values,
-                      ABCAdapter.KEY_DATATYPE: datatype}
-
-        if total_count > MAXIMUM_DATA_TYPES_DISPLAYED:
-            parameters[KEY_WARNING] = WARNING_OVERFLOW
-
-        if constants.ATT_REQUIRED in current_node:
-            parameters[constants.ATT_REQUIRED] = current_node[constants.ATT_REQUIRED]
-            if len(values) > 0 and string2bool(str(parameters[constants.ATT_REQUIRED])):
-                parameters[ABCAdapter.KEY_DEFAULT] = str(values[-1][ABCAdapter.KEY_VALUE])
-        previous_selected = self.context.get_current_default(name)
-        if previous_selected in [str(vv['value']) for vv in values]:
-            parameters[ABCAdapter.KEY_DEFAULT] = previous_selected
-
-        template_specification = {"inputRow": parameters, "disabled": False,
-                                  "parentDivId": parent_div, common.KEY_SESSION_TREE: tree_session_key}
-        return self.fill_default_attributes(template_specification)
+        name_start_index = datatype_index_path.rfind('.')
+        datatype_index_name = datatype_index_path[name_start_index+1:]
+        index_class = getattr(sys.modules[datatype_index_path[: name_start_index]], datatype_index_name)
+        filters_dict = json.loads(filters)
+
+        fields = []
+        operations = []
+        values = []
+
+        for idx in range(len(filters_dict['fields'])):
+            fields.append(filters_dict['fields'][idx])
+            operations.append(filters_dict['operations'][idx])
+            values.append(filters_dict['values'][idx])
+
+        filter = FilterChain(fields=fields, operations=operations, values=values)
+        project = common.get_current_project()
+
+        form = Form(project_id=project.id, draw_ranges=True)
+        data_type_gid_attr = DataTypeGidAttr(linked_datatype=REGISTRY.get_datatype_for_index(index_class()))
+        select_field = TraitDataTypeSelectField(data_type_gid_attr, form, conditions=filter)
+
+        options_response = ''
+        for option in select_field.options():
+            options_response = options_response + '<option value=' + '"' + option.value + '"' + ' class="form-control">' + option.label + '</option>'
```

Comment:
This HTML string concatenation is not good to have.
Can't you return a json which is parsed in the JS and converted in HTML code there on the client ?
Or even better use a Jinja template to render the response as HTML

Reply:
JSON would work while using some template, so should I create a new Jinja template for this?

Reply:
yes, either a new jinja template, and then you use @using_template or return a dict and then use annotation @jsonify on the controller method

Reply:
This part is solved.